src/ngx_http_lua_ctx.c

```diff
@@ -145,19 +151,67 @@ ngx_http_lua_ffi_get_ctx_ref(ngx_http_request_t *r)
 }
 
 
+int
+ngx_http_lua_ffi_get_ssl_ctx_ref(ngx_http_request_t *r, int *in_ssl_phase)
```

Comment:
I might be missing something, but it seems that we can change the existing `ngx_http_lua_ffi_get_ctx_ref` function to look for `ssl_ctx-> ssl_ctx` as well and return `int *need_clone` based on the current phase and where ref was located instead? That will save one FFI call and introducing new function.